src/Toolkit/Toolkit/GeoViewController.cs

```diff
@@ -0,0 +1,279 @@
+        /// </summary>
+        /// <param name="geoView"></param>
+        protected virtual void OnGeoViewAttached(GeoView geoView)
+        {
+        }
+
+        /// <summary>
+        /// Raised when the <see cref="GeoViewController"/> has been detached from a <see cref="GeoView"/>.
+        /// </summary>
+        /// <param name="geoView"></param>
+        protected virtual void OnGeoViewDetached(GeoView geoView)
+        {
+        }
+
+        /// <summary>
+        /// Raised when the attached <see cref="GeoView"/> loads into the active view.
+        /// </summary>
+        /// <param name="geoView">GeoView that was loaded</param>
+        protected virtual void OnGeoViewLoaded(GeoView geoView)
+        {
+        }
+
+        /// <summary>
+        /// Raised when the attached <see cref="GeoView"/> unloads from the active view.
+        /// </summary>
+        /// <param name="geoView">GeoView that was unloaded</param>
+        protected virtual void OnGeoViewUnloaded(GeoView geoView)
+        {
+        }
+
+        private void GeoView_Loaded(GeoView geoView)
+        {
+            Debug.Assert(geoView == ConnectedView && ConnectedView is not null);
+            OnGeoViewLoaded(geoView);
+        }
+
+        private void GeoView_Unloaded(GeoView sender)
+        {
+            Debug.Assert(sender == ConnectedView && ConnectedView is not null);
+            OnGeoViewUnloaded(sender);
+        }
+
+        #region Viewpoints
+
+        /// <inheritdoc cref="GeoView.GetCurrentViewpoint(ViewpointType)"/>
+        public virtual Viewpoint? GetCurrentViewpoint(ViewpointType viewpointType) => ConnectedView?.GetCurrentViewpoint(viewpointType);
+
+        /// <inheritdoc cref="GeoView.SetViewpointAsync(Viewpoint)"/>
+        public virtual Task<bool> SetViewpointAsync(Viewpoint viewpoint) => ConnectedView?.SetViewpointAsync(viewpoint) ?? Task.FromResult(false);
+
+        /// <inheritdoc cref="GeoView.SetViewpointAsync(Viewpoint, TimeSpan)"/>
+        public virtual Task<bool> SetViewpointAsync(Viewpoint viewpoint, TimeSpan duration) => ConnectedView?.SetViewpointAsync(viewpoint, duration) ?? Task.FromResult(false);
+
+        /// <inheritdoc cref="GeoView.SetViewpoint(Viewpoint)"/>
+        public virtual void SetViewpoint(Viewpoint viewpoint) => ConnectedView?.SetViewpoint(viewpoint);
+
+        #endregion Viewpoints
+
+        #region Identify
+        
+        /// <inheritdoc cref="GeoView.IdentifyLayersAsync(Point, double, bool, CancellationToken)"/>
+        public virtual Task<IReadOnlyList<IdentifyLayerResult>> IdentifyLayersAsync(Point screenPoint, double tolerance, bool returnPopupsOnly = false, CancellationToken cancellationToken = default) =>
+            ConnectedView?.IdentifyLayersAsync(screenPoint, tolerance, returnPopupsOnly, cancellationToken) ??
+                Task.FromResult<IReadOnlyList<IdentifyLayerResult>>(new List<IdentifyLayerResult>().AsReadOnly());
```

Comment:
There's a simpler way:
```suggestion
                Task.FromResult<IReadOnlyList<IdentifyLayerResult>>(Array.Empty<IdentifyLayerResult>());
```